### PR TITLE
merge upstream changes from codespecs

### PR DESCRIPTION
**Pull request by markro49**
This pull request includes changes to support Daikon on jdk11.  The version number is bumped to 6.4.2.  The last official Apache release is 6.4.1.  This version has three changes to support Daikon on jdk11:
- src/main/java/org/apache/bcel/generic/MethodGen.java  to improve Attribute support
- src/main/java/org/apache/bcel/generic/InstructionFactory.java to add support for static interfaces
- src/main/java/org/apache/bcel/generic/FieldGenOrMethodGen.java to improve Annotation support

It also contains one cosmetic change:
- src/main/java/org/apache/bcel/classfile/Code.java to improve the toString() output.

### CODE REVIEW

**Comment by mernst** [comment]
Could you change the version number?
Because this is built off Apache version 6.4.1, so the version number should be something like 6.4.1.0.1.  Apache might make other changes before their 6.4.2, and we want our 6.4.2 release to track that one.

**Review by mernst — APPROVED** [review]
This is fine to merge after fixing the version number.
To clarify, the commit comment mentions some changes, but those are all changes in its upstream -- this version is the same as its upstream (except for annotations), right?  I understand that the upstream codespects does differ from apache (which is codespecs's upstream).

**Comment by markro49** [comment]
I was aware of the issue – your suggestion is a good idea.

**Comment by markro49** [comment]
You are correct.

 

By the way, I would like to change the main working branch name from ‘trunk’ to ‘master’ to match both of the upstream repositories.  Not quite sure if that is a simple rename or more involved.  Can do after merge.

 

Also, what is status/plans for Javadoc on jdk11?

**Comment by mernst** [comment]
> Also, what is status/plans for Javadoc on jdk11?

Can you please clarify this question?

(There was a HTML error that I fixed, and now the CI job is passing.)

**Comment by markro49** [comment]
You fixed the problem/answered my question.  I'm working on the version number changes.

**Comment by mernst** [comment]
The Javadoc execution was confusing because there were dozens of "warning:" lines, but they didn't break the build (even with them, Javadoc still completes successfully).  Only the one "error:" line at the very end caused the problem.

**Comment by markro49** [comment]
I submitted a pull request to Apache to fix this upstream.

File: pom.xml
Comment (mernst) [inline pom.xml]: Please make sure that the buildfile runs the checkers.

**Review by mernst — CHANGES_REQUESTED** [review]

File: pom.xml
Comment (markro49) [reply pom.xml]: wasn't running checkers because i didn't do a mvn clean.  doh!

merge error is fixed

**Comment by mernst** [comment]
I suggest that each pull request should address one specific issue.
Please change the Checker Framework version and other dependencies in a different pull request, which can be merged either before or after this one.  That will make the changes easier to review, and it will make problems easier to debug.

File: pom.xml
Comment (mernst) [inline pom.xml:36]: This is not a change from upstream, so it should not be in this pull request.  It should be in a separate commit in typetools/commons-bcel, when we are ready to make a release.

File: pom.xml
Comment (mernst) [inline pom.xml:55]: This is not a change from upstream, so it should not be in this pull request.

File: src/main/java/org/apache/bcel/generic/Type.java
Comment (mernst) [inline src/main/java/org/apache/bcel/generic/Type.java:20]: This is a good change to bring typetools/commons-bcel in accordance with its upstream (thanks), but it isn't a change from the pull request.  It would have been clearer to fix gratuitous differences from upstream first, in a separate commit.  (But, once irrelevant changes like this are eliminated, they should never come back.)

The same comment applies to some other changes, like those in .gitignore.

File: src/main/java/org/apache/bcel/generic/ReferenceType.java
Comment (mernst) [inline src/main/java/org/apache/bcel/generic/ReferenceType.java]: This change is probably better style, but it is a gratuitous difference from upstream.  Please undo it.

**Review by mernst — CHANGES_REQUESTED** [review]

**Pull request merged**
